src/adapters/net.cpp

```diff
@@ -102,9 +105,67 @@ namespace net {
 
     freeifaddrs(ifaddr);
 
+    try {
+      this->query_ip6();
```

Comment:
Please reset the `m_ip6_last_error` after every successful ipv6 query so that we get warnings message after we lose the ipv6 address *again*

Reply:
Forgot about that basic part while I was busy overthinking this. :)

>on my machine, I get ::1 for the ipv6 address
That seems weird. That's loopback and should only be on your `lo` interface. These changes don't do anything interface specific (although they probably ought to). But it sounds like Linux is saying your IPv6 address is going to be sourced from your loopback address, which is weird.

Reply:
Yeah I'm not entirely sure what to do about the IPv6 address information showing up for all interfaces. It's almost better if the IPv6 thing is a separate module to do with routing or something that shows you what source address and interface it would use to get somewhere.

Reply:
Oh man! Totally didn't occur to me that this will give the same ipv6 address independent of the interface :frowning_face:
So this could only be used for generic ipv6 connectivity (without specifying the interface)
From what I've seen, to open a socket bound to a specific interface you need to have root and so I think the only option is to just use `getifaddrs` as we already do for ipv4 and use any of the ipv6 addresses on that interface. Technically that's what we do for ipv4 as well, because an interface can also have multiple ipv4 addresses.

Reply:
The problem with getifaddrs is that I kept getting the wrong address. Even if I filtered out link local and filtered by their scope. But curl seems to use getifaddrs successfully. They're probably doing something right that I missed. So I'll look at that later and if I figure it out I'll rewrite this.

Reply:
If an interface has multiple ipv6 addresses (with global scope), how can any of them be the *right* one?

From looking at curl [`lib/if2ip.c`](https://github.com/curl/curl/blob/master/lib/if2ip.c#L142), it filters addresses out by scope, but depending on the destination address (which we don't have).
How many ipv6 addresses does your interface have, that you are getting so many different addresses?
I would have said that it is a safe heuristic it to always use an address with global scope (if available) and otherwise choose any of the other addresses, because most of our users will have an ipv6 address with global scope (if they have an ipv6 address). But if you still pick the "wrong" address after that, then maybe it's not such a good heuristic.

Reply:
Turns out maybe avahi-daemon wasn't responsible for the extra addresses, maybe it was just logging about them innocently, because I stopped the service and rebooted and still have these extra weird addresses. I'm thinking that NetworkManager is doing something. But I'm not sure why. As far as I can tell, SLAAC shouldn't be enabled. With the previous change, polybar displays the wrong address. Something I might look at is what `ip route get to 2001:7fd::1 dev eno1` does, but netlink can be a bit gross to deal with.

Reply:
> The one I use to talk to the internet. :)

Yeah obviously, but isn't it possible that a single interface has multiple IPs with which it can reach the internet? On my Raspberry PI, I have three global ipv6 addresses on the same interface.

> Something I might look at is what `ip route get to 2001:7fd::1 dev eno1` does, but netlink can be a bit gross to deal with.

Yeah, or `ping -6 -I eno1 2001:7fd::1` will also give you an ipv6 address that can reach the internet.

Reply:
Just checked and I can ping `2001:7fd::1` with two of the three addressed. It seems we might have to inspect some routing tables in order to determine which IPv6 can reach the internet. Worst case, we can call `ip route get to 2001:7fd::1 dev eno1` and inspect the output